src/middle/Mir.ml

```diff
@@ -112,7 +109,8 @@ type 's fun_def =
   ; fdloc: location_span sexp_opaque [@compare.ignore] }
 [@@deriving sexp, hash, map]
 
-type 'e lvalue = string * 'e index list [@@deriving sexp, hash, map, fold]
+type 'e lvalue = string * unsizedtype * 'e index list
+[@@deriving sexp, hash, map, fold]
```

Comment:
Right. I can see that it would be useful to have the type information here. Another option there is to go the full recursive route that I ended up using in the AST, where lvalues just have the same metadata as expressions, at each recursive nesting. That has the advantage that it is very easy to convert back and forth between expressions and lvalues.

On the other hand, based on the discussion here: https://github.com/stan-dev/stan/issues/2795 , it seems like that sort of recursive lvalue is something we might want to disallow anyway in the future?

Maybe we can discuss this on the phone?